agent-control/src/event/cancellation.rs

```diff
@@ -5,15 +5,23 @@ use std::time::Duration;
 pub type CancellationMessage = ();
 
 impl EventConsumer<CancellationMessage> {
-    /// Check if the consumer is cancelled.
+    /// Checks whether the consumer is cancelled immediately.
+    ///
+    /// Calls [`Self::is_cancelled`] with a timeout of zero.
+    pub fn is_cancelled(&self) -> bool {
+        self.is_cancelled_with_timeout(Duration::ZERO)
+    }
+
+    /// Checks whether the consumer is cancelled for the given timeout.
+    ///
     /// It returns true if the consumer received a cancellation message or received an error
     /// before the provided timeout is elapsed. Otherwise it blocks until the timeout is elapsed
     /// and returns false.
-    pub fn is_cancelled(&self, timeout: Duration) -> bool {
-        let timed_out = matches!(
-            self.as_ref().recv_timeout(timeout),
-            Err(RecvTimeoutError::Timeout)
-        );
-        !timed_out
+    pub fn is_cancelled_with_timeout(&self, timeout: Duration) -> bool {
+        match self.as_ref().recv_timeout(timeout) {
+            Ok(_) => true,
+            Err(RecvTimeoutError::Timeout) => false,
+            Err(RecvTimeoutError::Disconnected) => true,
```

Comment:
I think this might work

```suggestion
        match self.as_ref().recv_timeout(timeout) {
            Ok(_) | Err(RecvTimeoutError::Disconnected) => true,
            Err(RecvTimeoutError::Timeout) => false,
            
```